### PR TITLE
Docs Search 

### PR DESCRIPTION
**Pull request by tommy-56**
Draft PR for Docs Search. 
Things to do/ could use some guidance
- github actions for algolia gastby plugin

### CODE REVIEW

File: package.json
Comment (alanchrt) [inline package.json]: I think one of the main pieces of work going forward on this will be to eliminate any styled-component dependencies, since we use a different styling solution (material-ui v4 and step-ui). That may be worth starting to dig into... how you will rewrite the included components based on styled-components using material-ui and step-ui. Will work offline to help you get set up on step-ui! Here are the docs for material-ui. Be sure to reference v4 docs, not v5 (we're not upgraded yet):

https://v4.mui.com/

File: package.json
Comment (alanchrt) [inline package.json]: This dependency will end up going away with the port to material-ui and step-ui as well.

File: src/components/Search/search-result.jsx
Comment (alanchrt) [inline src/components/Search/search-result.jsx]: `<ListItem>` (from `@material-ui/core`)

Comment (alanchrt) [reply src/components/Search/search-result.jsx]: https://v4.mui.com/components/lists/

File: src/components/Search/search-result.jsx
Comment (alanchrt) [inline src/components/Search/search-result.jsx]:
```javascript

const useStyles = makeStyles(theme => ({
  hits: {
    'ul': {
      listStyle: 'none',
      ...
    }
    'li.ais-Hits-item': {
      marginBottom: '1em',
      'a': {
        color: theme.foreground,
        ....
      }
    }
  }
});

...

const classes = useStyles();

...

<Hits className={classes.hits} ... >

...
```

**Docs:** https://v4.mui.com/styles/basics/#hook-api

File: src/components/Search/search-box.jsx
Comment (alanchrt) [inline src/components/Search/search-box.jsx]:
```suggestion
        width={350}
```

**Comment by tommy-56** [comment]
The styled-components dependencies are gone and its fully using step-ui/mui and is fully functional
Looking for the next round of feedback  👍 @alanchrt

File: gatsby-config.js
Comment (alanchrt) [inline gatsby-config.js:1]: I think this should be ok. Do we use `.env` files in any other environments? I'm pretty sure we don't. :+1: 

Could you delete the commented out code?

File: package.json
Comment (alanchrt) [inline package.json]: Is this still needed? We should use all material icons, so could we uninstall this dependency?

File: package.json
Comment (alanchrt) [inline package.json]: Is this specified somewhere as being needed to explicitly install for the plugin? Can we uninstall it?

File: package.json
Comment (alanchrt) [inline package.json]: Can we uninstall this one?

File: package.json
Comment (alanchrt) [inline package.json]: Can we uninstall the `hooks-web` one?

File: package.json
Comment (alanchrt) [inline package.json:46]: Could you delete the `package-lock.json` file from the repo? We use `yarn.lock` to lock dependency versions, but package-lock is for npm instead of yarn. :+1:

File: src/algolia-queries.js
Comment (alanchrt) [inline src/algolia-queries.js]: Does this mean we'll only be able to search the first 3k characters of each page rather than the full `body`? I see they have a 10KB limit for index items, which is a bummer. But, maybe we can at least make this a little higher since we don't have much else in the index? 8k?

File: src/components/Search/index.jsx
Comment (alanchrt) [inline src/components/Search/index.jsx]: Delete this one. :+1: (wouldn't work as expected here anyway)

File: src/components/Search/index.jsx
Comment (alanchrt) [inline src/components/Search/index.jsx]: The convention here is to name this `classes`, not `theme`, since it's the generated CSS class names for the styles (eg. in `classes.root`. `theme` is actually a different concept in material-ui, referring to the styling context across all components.

File: src/components/Search/index.jsx
Comment (alanchrt) [inline src/components/Search/index.jsx]: I think this wrapper shouldn't be needed since we already apply it globally to the whole app. But, similar to what happened above, it looks like you're actually just trying to apply a CSS background. Maybe just make this `<div className={classes.root}?` instead?— after you rename `theme` to `classes`.

File: src/components/Search/index.jsx
Comment (alanchrt) [inline src/components/Search/index.jsx]: Is `border={1}` needed? Can this just be a div with `theme.resultsBorder` applied? Also, it would be good to avoid one-off colors. Maybe I can try to get this running locally to see it and make a recommendation.

File: src/components/Search/index.jsx
Comment (alanchrt) [inline src/components/Search/index.jsx]: Do these styles actually apply successfully when passed this way? I think I'd expect them to live in the `SearchResult` component/file instead. Can we move them there?

File: src/components/Search/search-result.jsx
Comment (alanchrt) [inline src/components/Search/search-result.jsx]: I'm wondering where we add borders if the `Paper` component would be more appropriate. Will try to get it running and see!

File: src/components/Search/search-result.jsx
Comment (alanchrt) [inline src/components/Search/search-result.jsx]: Nice you guys got this to work! I think you can delete the `hit` and `tagName` props here.

File: src/components/Search/search-result.jsx
Comment (alanchrt) [inline src/components/Search/search-result.jsx:51]: Confused. What is this check doing? We're determining URL based on whether the title is all caps?

**Review by alanchrt — CHANGES_REQUESTED** [review]
I left some smaller feedback, but this is really great work! Amazing you guys rolled this altogether as a first substantial React project.

I still haven't run it locally, so I'm going to try to get that going. So, expect a little more feedback on visual tweaks and stuff. But, overall, this is fantastic. Excited to see it ship.

:tada:

File: package.json
Comment (masoncooper19) [reply package.json]: None of the styled stuff should be needed, we can get rid of that dependency!

File: src/algolia-queries.js
Comment (tommy-56) [reply src/algolia-queries.js]: That is exactly what it is doing.  I believe we had it set much higher about 6k before without issue so we can bump it up a bit 👍

File: src/components/Search/search-result.jsx
Comment (tommy-56) [reply src/components/Search/search-result.jsx:51]: The slugs for the step cli commands are stored like this. The command step beta, for example, the slug is just '/beta' so we had to find a way to append `step-cli/reference/` to the slug. Since the titles of all the cli commands are lowercase this was our way of checking if the search result was a cli command or not.

**Comment by masoncooper19** [comment]
@alanchrt we made some changes to the original tommy-mason doc search branch, we changed things based on your feedback in both pr's. We tried in the past to hide the navbar and just have the list go over but after receiving other feedback from Josh we instead turned our list into a popover essentially. We can try to mess around more with the popover on the other branch you created but I'm having a feeling we're going to be getting a lot more scaling issues that I'm not sure we'd be able to resolve very easily. What do you think?

File: src/components/Search/search-result.jsx
Comment (alanchrt) [reply src/components/Search/search-result.jsx:51]: Ahh, I see. I think they may need to be stored differently if we don't know how to construct the URLs from the data in the index. This will be unreliable and fickle if we end up having a page that has a slug matching this condition unexpectedly.

Is there a way you can just store the full path to individual pages when you store the data in Algolia, then use that path here instead of this workaround?

File: src/components/Search/index.jsx
Comment (alanchrt) [reply src/components/Search/index.jsx]: This one still needs to be updated to `classes`. :+1:

**Review by alanchrt — CHANGES_REQUESTED** [review]
Recent update looks good! I'd say once the search results are flattened into the side nav (kill the popover), this should be good to merge. Let me know if you need some help on how to hide navigation, since it's in the containing component for search, not nested inside it.

**Comment by alanchrt** [comment]
Example:

search-inline

File: src/components/Search/search-result.jsx
Comment (tommy-56) [reply src/components/Search/search-result.jsx:51]: After playing around today and a lot the other day, I still can't seem to find a way to get a longer version of the slugs into Algolia. I was looking around on the GraphiQL and saw some other graphql queries that would grab the right slug but when I tried using them to talk to Algolia it would throw me an error from Algolia since it did not recognize certain attributes. 
There is an attribute that takes in the full file path but there would still need to be a bit of a work around to trim down all the unnecessary file path info.

I may just be missing some knowledge on Graphql in general or something but for me I could not find a good way to get better slugs into Algolia.

Comment (alanchrt) [reply src/components/Search/search-result.jsx:51]: Ok, that stinks, but I don't think it's worth holding up shipping. Once everything else is fully ready for review, let's go ahead and get a last look at it, then our team can come back around and see if we get to fixing this. I made #120 so we don't lose track. :)

**Comment by tashian** [comment]
This PR is my favorite

File: src/components/Search/index.jsx
Comment (alanchrt) [inline src/components/Search/index.jsx]: This border can be deleted now that the results are inline.

```suggestion
```

And then the corresponding `<Box>` can go away too.

File: src/components/Search/index.jsx
Comment (alanchrt) [inline src/components/Search/index.jsx]: This is the wrapping `<Box>` that can just be deleted.

File: src/components/Search/search-result.jsx
Comment (alanchrt) [inline src/components/Search/search-result.jsx]: Smaller size is good here.

```suggestion
      <Typography variant="body2">
```

**Review by alanchrt — APPROVED** [review]
@masoncooper19 @tommy-56 I pulled this down and poked around and it looks great! I added a couple very minor style tweaks. Once those are committed, feel free to merge away!

Awesome work. :boom:

**Pull request merged**
